src/ConnectedMode/Binding/BoundConnectionInfoProvider.cs

```diff
@@ -55,15 +53,11 @@ public IEnumerable<BoundConnectionInfo> GetExistingBindings()
 
             var result = new List<BoundConnectionInfo>();
 
-            var bindings = unintrusiveBindingPathProvider.GetBindingPaths();
+            var bindings = solutionBindingRepository.List();
 
             foreach (var binding in bindings)
             {
-                var boundSonarQubeProject = solutionBindingFileLoader.Load(binding);
-
-                if (boundSonarQubeProject == null) { continue; }
-
-                result.Add(ConvertToBindingInfo(boundSonarQubeProject));
+                result.Add(ConvertToBindingInfo(binding));
```

Comment:
You don't need to have the conversion if you can directly read this class from json